src/deploy/functions/cache/applyHash.ts

```diff
@@ -34,13 +36,28 @@ export async function applyBackendHashToBackends(
 function applyBackendHashToEndpoints(
   wantBackend: Backend,
   envHash: string,
+  codebaseFilters: EndpointFilter[],
   sourceV1Hash?: string,
   sourceV2Hash?: string
 ): void {
   for (const endpoint of allEndpoints(wantBackend)) {
     const secretsHash = getSecretsHash(endpoint);
     const isV2 = endpoint.platform === "gcfv2";
     const sourceHash = isV2 ? sourceV2Hash : sourceV1Hash;
-    endpoint.hash = getEndpointHash(sourceHash, envHash, secretsHash);
+    // If the endpoint is in the filtered list, then skip setting a hash (effectively forcing a deploy).
+    if (!isEndpointFiltered(endpoint.id, codebaseFilters)) {
+      endpoint.hash = getEndpointHash(sourceHash, envHash, secretsHash);
+    }
   }
 }
+
+function isCodebaseFiltered(codebase: string, filters: EndpointFilter[]) {
```

Comment:
This looks like it could live in `functionsDeployHelper.ts`, no?

Reply:
Good point, moved it!